.github/workflows/build-and-test.yaml

```diff
@@ -0,0 +1,30 @@
+name: Build and Test
+
+on: [push]
+
+jobs:
+  build:
+    
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go-version: [ '1.17', '1.19' , '1.20' ]
```

Comment:
Why so many different versions?

Reply:
What to test backward compatibility with previous golang versions